lib/stacks/eks-blueprint-stack.ts

```diff
@@ -57,14 +57,20 @@ export class EksBlueprint extends cdk.Stack {
         const vpcId = this.node.tryGetContext("vpc");
         const vpc = this.initializeVpc(vpcId);
 
-        const clusterProvider = blueprintProps.clusterProvider ?? new EC2ClusterProvider;
+        const clusterProvider = blueprintProps.clusterProvider ?? new EC2ClusterProvider();
 
         const clusterInfo = clusterProvider.createCluster(this, vpc, blueprintProps.version ?? KubernetesVersion.V1_19);
 
         const postDeploymentSteps = Array<ClusterPostDeploy>();
 
         for (let addOn of (blueprintProps.addOns ?? [])) { // must iterate in the strict order
-            addOn.deploy(clusterInfo);
+            const node = addOn.deploy(clusterInfo);
+            if (node instanceof cdk.Construct) {
+                if (clusterInfo.provisionedAddOns == undefined) {
```

Comment:
Does it buy us anything that `ClusterInfo` is an interface? It seems like we should make it a class and have the `provisionedAddons` initialized in the constructor so that we don't have to test for `null` before access.
If we make it a class, we could just add a method like `clusterInfo.addProvisionedAddOn(addon, construct)` and `getProvisionedAddon(addonName)` to make the client code simpler.

Happy to discuss if you see and cons for this approach.

Reply:
I don't think it does buy us anything. I was thinking the same thing when I was implementing this part but thought it might be a breaking change. Turns out its not that bad and I have made it into a class.